tests/escapes.rs

```diff
@@ -0,0 +1,126 @@
+use html_editor::operation::*;
+use html_editor::{parse, Node, Element};
+
+#[test]
+fn test_parse() {
+    const HTML: &str = r#"
+        <!DOCTYPE html>
+        <html lang="en">
+        <head>
+            <meta charset="UTF-8">
+            <title>I &lt;3 &#34;escaping&#34;</title>
+        </head>
+        <body>
+            <div id="testee" attr="id-with-&quot;quotes&quot;-inside"></div>
+        </body>
+        </html>"#;
+
+    let html = parse(HTML).unwrap();
+    let title_selector = Selector::from("title");
+
+    let Some(title) = html.query(&title_selector) else {
+        assert!(false, "Invalid title");
+        return;
+    };
+
+    assert_eq!(title.name, "title");
```

Comment:
How about improve it like this:
```rust
match html.query(&title_selector) {
    Some(title) => assert_eq!(title.name, "title");
    None => assert!(false);
}
```
I think this one would be more clearer.

Same for the code below.

Reply:
Problem is, `title` is used later in the test, meaning one would have to return `title` in the first branch. However, as `assert!(...)` returns a (), we would have to add a `return` or `unreachable!()` in the `None`-case.

Reply:
Oops I didn't notice it.

Reply:
Oops I didn't notice it.